test/test_transforms_v2_refactored.py

```diff
@@ -1697,3 +1702,185 @@ def call_transform():
+                        factor = (input_max_value + 1) // (output_max_value + 1)
+                        return value / factor
+                    else:
+                        factor = (output_max_value + 1) // (input_max_value + 1)
+                        return value * factor
+
+        return torch.tensor(tree_map(fn, image.tolist()), dtype=dtype)
+
+    @pytest.mark.parametrize("input_dtype", [torch.float32, torch.float64, torch.uint8, torch.int64])
+    @pytest.mark.parametrize("output_dtype", [torch.float32, torch.float64, torch.uint8, torch.int64])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    @pytest.mark.parametrize("scale", (True, False))
+    def test_image_correctness(self, input_dtype, output_dtype, device, scale):
+        if input_dtype.is_floating_point and output_dtype == torch.int64:
+            pytest.xfail("float to int64 conversion is not supported")
+
+        input = make_image(dtype=input_dtype, device=device)
+
+        out = F.to_dtype(input, dtype=output_dtype, scale=scale)
+        expected = self.reference_convert_dtype_image_tensor(input, dtype=output_dtype, scale=scale)
+
+        if input_dtype.is_floating_point and output_dtype is torch.uint8 and scale:
+            torch.testing.assert_close(out, expected, atol=1, rtol=0)
+        else:
+            torch.testing.assert_close(out, expected)
+
+    @pytest.mark.parametrize(
+        "make_input",
+        (
+            make_image_tensor,
+            make_image,
+            make_video,
+        ),
+    )
+    def test_behaviour(self, make_input):
+        def was_scaled(inpt):
+            # this assumes the target dtype is float
+            return inpt.max() <= 1
+
+        H, W = 10, 10
+        sample = {
+            "img": make_input(size=(H, W), dtype=torch.uint8),
+            "bbox": make_bounding_box(size=(H, W)),
+            "mask": make_detection_mask(size=(H, W)),
+        }
+
+        img_dtype = sample["img"].dtype
+
+        bbox_dtype = sample["bbox"].dtype
+        assert bbox_dtype == torch.float32
+
+        mask_dtype = sample["mask"].dtype
+        assert mask_dtype == torch.bool
+
+        # only img gets transformed when dtype isn't a dict
+        for scale in (True, False):
+            out = transforms.ToDtype(dtype=torch.float32, scale=scale)(sample)
+            assert out["img"].dtype == torch.float32
+            if scale:
+                assert was_scaled(out["img"])
+            else:
+                assert not was_scaled(out["img"])
+            assert out["bbox"].dtype == bbox_dtype
+            assert out["mask"].dtype == mask_dtype
+
+        # make sure "others" works as a catch-all and that None means no conversion
+        out = transforms.ToDtype(dtype={datapoints.Mask: torch.int64, "others": None})(sample)
+        assert out["img"].dtype == img_dtype
+        assert out["bbox"].dtype == bbox_dtype
+        assert out["mask"].dtype != mask_dtype
+        assert out["mask"].dtype == torch.int64
```

Comment:
They are redundant but I slightly prefer keeping both, because just reading

```
assert out["mask"].dtype == torch.int64
```

doesn't make it obvious that the output dtype is different from the input dtype